### PR TITLE
Call newsletter API rather than embed an iframe

### LINKED ISSUE #569: Replace embedded mailjet newsletter form with an API call and a fresh UI
Right now we have a mailjet form that works via embedded iframe that attempts to resize itself, and it doesn't always work well:

image

goal: replace this iframe in [NewsletterForm.js](https://github.com/apisyouwonthate/apisyouwonthate.com/blob/main/src/components/NewsletterForm/NewsletterForm.js) with some nice UI that calls the [Mailjet Contact API](https://dev.mailjet.com/email/guides/contact-management/#single-contact-management) directly, likely via an API route in this project (`/api/newsletter` etc)

### PR DESCRIPTION
**Pull request by mbifulco**
fixes #569

image

### CODE REVIEW

File: src/pages/api/newsletter/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// attempt to subscribe to newsletter
+export default async function handler(req, res) {
+  if (req.method !== 'POST') {
+    res.status(400).send({ message: 'Only POST requests allowed' });
```
Comment (philsturgeon) [inline src/pages/api/newsletter/index.js:4]:
```suggestion
    res.status(405).send({ message: 'Only POST requests allowed' });
```

**Comment by mbifulco** [comment]
@philsturgeon can you give this a look and a test? I haven't used mailjet to send any emails, so I'm not sure if there's additional work that needs to be done to make sure people who sign up with this form will receive newsletters.  I noticed that 
 https://app.mailjet.com/contacts/subscribers/contact_list has this strange warning:

image

So I reworked the code to [create a contact](https://dev.mailjet.com/email/guides/contact-management/#create-a-contact) and then [add them to a list](https://dev.mailjet.com/email/guides/contact-management/#add-a-contact-to-a-contact-list).

The API code in `/api/newsletter/index.js` feels a little gross because of the odd 2-step subscription process, but I'm not to lose much sleep over it 🤷🏼

File: src/pages/api/newsletter/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        IsUnsubscribed: 'false',
+        ContactID: id,
+        ContactAlt: email,
+        ListID: process.env.MAILJET_NEWSLETTER_LIST_ID,
+      });
+    res.status(200).send(listRequest.body);
+  } catch (err) {
+    if (
+      err.message.startsWith(
+        'Unsuccessful: 400 A duplicate ListRecipient already exists.'
```
Comment (philsturgeon) [inline src/pages/api/newsletter/index.js:52]: Maybe if they try adding themselves to the list and they're already on the list we should consider that a success, and only complain if its some other sort of error?

File: src/pages/api/newsletter/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    res.status(err.statusCode).send({ message: err.message });
+  }
+}
+
+const addToList = async ({ id, email }, res) => {
+  try {
+    const listRequest = await mailjet
+      .post('listrecipient', { version: 'v3' })
+      .request({
+        IsUnsubscribed: 'false',
```
Comment (philsturgeon) [inline src/pages/api/newsletter/index.js:43]:
```suggestion
```

**Comment by philsturgeon** [comment]
@mbifulco yeah by looking at the /listrecipient endpoint it looks like we're doing what we should be. 

https://dev.mailjet.com/email/reference/contacts/subscriptions/#v3_post_listrecipient

Nicely done!

File: src/pages/api/newsletter/index.js (hunk 2 above)
Comment (mbifulco) [reply src/pages/api/newsletter/index.js:52]: Yep, that's what we're doing here 👍🏼  On the web-side, this gives the same ✅ _You did it!_ experience as a brand new subscriber would see.

**Pull request merged**